tests/rest/admin/test_room.py

```diff
@@ -1792,6 +1795,134 @@ def test_get_joined_members_after_leave_room(self) -> None:
         self.assertEqual(403, channel.code, msg=channel.json_body)
         self.assertEqual(Codes.FORBIDDEN, channel.json_body["errcode"])
 
+    def test_topo_token_is_accepted(self) -> None:
+        user = self.register_user("foo1", "pass")
+        user_tok = self.login("foo1", "pass")
+        room_id = self.helper.create_room_as(user, tok=user_tok)
+
+        token = "t1-0_0_0_0_0_0_0_0_0"
+        channel = self.make_request(
+            "GET",
+            "/_synapse/admin/v1/rooms/%s/messages?access_token=%s&from=%s"
+            % (room_id, self.admin_user_tok, token),
+        )
+        self.assertEqual(HTTPStatus.OK, channel.code)
+        self.assertTrue("start" in channel.json_body)
+        self.assertEqual(token, channel.json_body["start"])
+        self.assertTrue("chunk" in channel.json_body)
+        self.assertTrue("end" in channel.json_body)
+
+    def test_stream_token_is_accepted_for_fwd_pagianation(self) -> None:
+        user = self.register_user("foo2", "pass")
+        user_tok = self.login("foo2", "pass")
+        room_id = self.helper.create_room_as(user, tok=user_tok)
+
+        token = "s0_0_0_0_0_0_0_0_0"
+        channel = self.make_request(
+            "GET",
+            "/_synapse/admin/v1/rooms/%s/messages?access_token=%s&from=%s"
+            % (room_id, self.admin_user_tok, token),
```

Comment:
```suggestion
            "/_synapse/admin/v1/rooms/%s/messages?from=%s"
            % (room_id, self.admin_user_tok, token),
            access_token=self.admin_user_tok
```

IMO the access_token should not be part of the URL for security reason (e.g. exposing in log files).